awscli/utils.py

```diff
@@ -129,7 +130,9 @@ def _create_fetcher(self):
         metadata_num_attempts = self._session.get_config_variable(
             'metadata_service_num_attempts')
         imds_config = {
```

Comment:
I think we're missing [`ec2_metadata_service_endpoint`](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L70) in the `imds_config`. This allows users to set a custom IMDS endpoint directly which maintains precedence over setting the endpoint mode. For more background, it is resolved [here](https://github.com/boto/botocore/blob/develop/botocore/utils.py#L381).

Reply:
It looks like this was never added as part of this PR: https://github.com/aws/aws-cli/pull/5844. Given v2 has never used this variable for IMDS region resolution, I'd prefer to just send this update as a separate PR since this commit is only needed because it unbreaks the merge from the develop branch.

Reply:
Talked offline. We are going to send this update as a separate PR in order to unblock this one.